### PR TITLE
Site Settings: Introduce a new Protect card

### PR DESCRIPTION
**Pull request by tyxla**
### Description

This PR introduces a new version of the Protect card, aligned with the requirements in p6TEKc-HQ-p2. The legacy Protect card was not compliant with the latest UI and code requirements, and was in need of a substantial refactor, so it made sense to create a new card instead.

The PR is part of #11676.

### Technical implementation
This PR suggests the following changes and additions:

**State**
* Normalize the `jetpack_protect_global_whitelist` field to a simple string.
* Filter Protect module settings when saving settings.
* Tests for the above

**UI**
* Introduce the new Protect card.
* Add the new Protect card to the Security form.
* Remove the legacy Jetpack Protect card and the related README material.

### Preview

**(Protect module active, current IP not in whitelist):**

**(Protect module active, current IP in whitelist):**

**(Protect module deactivated):**

### To test

* Checkout this branch
* Go to `/settings/security/$site` where `$site` is a Jetpack site.
* Enable the Protect module from the module toggle in the new card (if it's not enabled).
* Play with the IP field and the "Add to Whitelist" button and try to break them.
* Verify the "Add to Whitelist" button is disabled when the current IP is already whitelisted.
* Verify the IPs save correctly in the database.
* Verify all tests still pass:

```
npm run test-client client/state/jetpack/settings/test/utils.js
```

### CODE REVIEW

File: client/my-sites/site-settings/protect.jsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		return (
+			<div>
+				<QueryJetpackConnection siteId={ selectedSiteId } />
+
+				<Card className="protect__card site-settings__security-settings">
+					<FormFieldset>
+						<div className="protect__info-link-container site-settings__info-link-container">
+							<InfoPopover position={ 'left' }>
+								<ExternalLink href={ 'https://jetpack.com/support/protect' } target="_blank">
+									{ translate( 'Learn more about Jetpack protect' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/protect.jsx]: Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Learn more about Jetpack. ' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by roccotripaldi** [comment]
>  I get a green success message even though the settings do not save.

This is also happening in wp-admin on the Jetpack site. I think it is an issue on WP REST API in Jetpack: https://github.com/Automattic/jetpack/blob/master/_inc/lib/class.core-rest-api-endpoints.php#L2072

**Comment by MichaelArestad** [comment]
I can't test fully since for some reason it can't detect my IP, but in that scenario, it worked perfectly by removing the button and saying it's unknown.

I think this looks and probably works well. That said, in JP, we shifted to a foldable card like this:

image

**Comment by oskosk** [comment]
Probably we need to add an ending period at the end of the description if it makes sense as there an IP address at the end of the sentence. Also, I thinkg that a period at the end of the InfoPopover text will be needed.

Looks, like the pattern matching in`isIpAddressWhitelisted` is not working if there's a range of IP addresses. I understand the calculation could be complicated, but it doesn't work even when the address is the beginning of a range. (On the screenshot you can see a range entered which includes the verbatim IP address.


Button disabled:

image

Button not disabled

image



Otherwise, LGTM. :shipit: 

Let's merge this and track the observations in a differente issue :
1. No error for invalid input from the API, mentioned by Rocco
2. Detection of user's IP address, mentioned by Michael

**Comment by tyxla** [comment]
> Probably we need to add an ending period at the end of the description if it makes sense as there an IP address at the end of the sentence. Also, I thinkg that a period at the end of the InfoPopover text will be needed.

Let's ask @MichaelArestad @rickybanister @ashleighaxios about that.

> Looks, like the pattern matching inisIpAddressWhitelisted is not working if there's a range of IP addresses. I understand the calculation could be complicated, but it doesn't work even when the address is the beginning of a range. (On the screenshot you can see a range entered which includes the verbatim IP address.

I've altered the calculations now, so if the IP is in the low or high of a range, it'll also consider that the IP is already whitelisted. We can also complicate it even more to always be accurate if the IP is in the middle of a range, but I don't think it's worth it.

@oskosk this one could use another code review.

I've also updated the card to use a `FoldableCard`. Here are some updated previews:

Collapsed: 


Expanded: 


@MichaelArestad @rickybanister @ashleighaxios this could use another design review, too.

**Comment by rickybanister** [comment]
This is really minor, but I notice the margin between the text and the 'add to whitelist' button is smaller in Calypso than Jetpack. I think the margin in Jetpack looks more natural there. It could use another couple pixels of space.

File: client/my-sites/site-settings/protect.jsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		return (
+			<FoldableCard className="protect__foldable-card site-settings__foldable-card" header={ protectToggle }>
+				<QueryJetpackConnection siteId={ selectedSiteId } />
+
+				<FormFieldset>
+					<div className="protect__module-settings site-settings__child-settings">
+						<div className="protect__info-link-container site-settings__info-link-container">
+							<InfoPopover position={ 'left' }>
+								<ExternalLink href={ 'https://jetpack.com/support/protect' } target="_blank">
+									{ translate( 'Learn more about Jetpack protect' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/protect.jsx]: Hi! I've found a possible matching string that has already been translated **30** times:
`translate( 'Learn more about Jetpack. ' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by rickybanister** [comment]
@MichaelArestad can you spin up a PR to get the copy here synced up with the changes you made to Jetpack yesterday. Go forward we should try to open issues or PRs simultaneously to keep things in sync.

**Comment by oskosk** [comment]
LGTM as it works as expected and code looks good.

**Comment by tyxla** [comment]
@rickybanister I've added 3px of spacing above the button:

Before:


After:

**Comment by rickybanister** [comment]
Looks great, ship it.

**Pull request merged**